### ISSUE TITLE
MongoDatabase>>#collections returns empty Collection

### ISSUE THREAD

**Issue by krodelin** [pharo-nosql/mongotalk#12]
`MongoDatabase>>#collections` uses `<database>.system.namespaces` to get the DB collections. However this has been deprecated as of  MongoDB 3.0 and thus returns an empty collection (https://docs.mongodb.com/manual/reference/system-collections/#%3Cdatabase%3E.system.namespaces).

The new was ist to use `listCollections` (https://docs.mongodb.com/manual/reference/command/listCollections/).

The following snippet uses `lisrtCollections`to retrieve the collection names. It just uses plain `MongoDatabase>>#command` though w/o any decoration:

```
db := Mongo default databases first.
reply := db command: { #listCollections -> 1 } asDictionary.
((reply at: 'cursor') at: 'firstBatch') collect: [ :each | each at: #name ].
```

**Comment by krodelin** [comment]
The following monkeypatch for `MongoDatabase>>#collections` can be used meanwhile:

```
collections
    | query reply names real stripped |
    "query := MongoQuery new collection: (MongoMetaCollection name: 'system.namespaces').
    reply := self query: query."
    "Drop options on the floor for now"
    "names := reply collect: [:each | each at: 'name']."
    reply := self command: { #listCollections -> 1 } asDictionary.
    names :=  ((reply at: 'cursor') at: 'firstBatch') collect: [ :each | each at: #name ].
    "real := names select: [:each | (each occurrencesOf: $.) = 1].
    stripped := real collect: [:each | each readStream upTo: $.; upToEnd]."
    ^names collect: [:each | MongoCollection database: self name: each]
```

**Comment by tinchodias** [comment]
the 2 failing tests are green with the "monkeypatch"

**Comment by tinchodias** [comment]
Do we still support Mongo older than 3?

If we do... One possibility is to go more low-level in MongoDatabase: remove "collections" and add "systemNamespaces" and "listCollections". Then, the called should know what to send.

Then, the 2 failing tests that need to know the number of collections can check "mongo majorVersion >= 3" and choose what to send. 

May be better to do not remove "collections" but reimplement it with the "mongo majorVersion >= 3" inside, and then return either "self systemNamespaces" or "self listCollections".

**Comment by tinchodias** [comment]
Additionally, we should implement cursors so listCollections can return the collections beyond firstBatch. Cursors can be useful for other commands.

**Comment by tinchodias** [comment]
Fixed with #27
